pkg/filestore/writer.go

```diff
@@ -74,6 +75,28 @@ func (w *Writer) WriteKeypair(meta metadata.Metadata, key crypto.PrivateKey, cha
 
 	keyPEM := pem.EncodeToMemory(pemBlock)
 
+	files := map[string][]byte{
+		attrs[csiapi.KeyFileKey]:  keyPEM,
+		attrs[csiapi.CertFileKey]: chain,
+		attrs[csiapi.CAFileKey]:   ca,
+	}
+
+	if attrs[csiapi.KeystoreFile] != "" {
+		switch keyStoreType := attrs[csiapi.KeystoreType]; keyStoreType {
```

Comment:
The 'if' here is asserting a string value is not false I think? Can you make this use a `, ok` second return parameter and check that instead?

Reply:
Also currently if keystoreType is not set, we silently don't return an error with the current code (and don't write any data). This should be an error instead of just being silent 😊 (+ a test case for this would be great!)